### PR TITLE
Service changes for results view

### PR DESCRIPTION
**Pull request by drothlis**
Misc. changes extracted from #25.

**TODO:**

- [x] Write a test checking the fields we get back from `/api/_private/results`
- [x] Remove "Conflicts" from commit messages

### CODE REVIEW

**Comment by drothlis** [comment]
@wmanley is 1092af9 still necessary, and why is it WIP?

**Comment by drothlis** [comment]
All the docker-requiring tests pass too, as of 8a64acf.

**Comment by drothlis** [comment]
> is 581227a still necessary, and why is it WIP?

@wmanley says it looks necessary, useful, and no longer WIP; so I edited the commit message and rebased. It's 51b0080 now ("results parser: If we don't know the job_uid, then make one up").

**Comment by drothlis** [comment]
Rebased onto master. @wmanley this is ready for review.

**Comment by wmanley** [comment]
328bb33b419dd8f3416c8a23353adcb797d2cfcd: Commit message says "Conflicts:" in it

**Comment by wmanley** [comment]
ed88a47de57730a0b69a929446b21783c1a172cd: Commit message says:

    Conflicts:
    	http-service/service.py
    	tests/test_http_service.py
    	web-ui/app/results.js
    	web-ui/app/views/results.html

**Comment by wmanley** [comment]
a4b67e0cecdf22d0e543d7a60749b6870f0ae247: Says:

    Conflicts:
    	http-service/service.py

**Comment by wmanley** [comment]
The commit messages are a bit sparse, and it'd be nice to have a test which checks which fields are being returned.  But I guess we *could* add them later,

File: http-service/service.py [hunk 1]
```diff
@@ -1046,18 +1052,31 @@ def load():
...
+            self, filtertext="", sort="date:desc", index=0, count=0):
+        fields = {
+            '_id': False,
+            'start_timestamp': True,
+            'test_run.exit_status': True,
+            'test_run.failure_reason': True,
+            'test_run.test_case.name': True,
+            'duration': True,
+            'id': True,
+        }
```
Comment (wmanley) [inline http-service/service.py]: I would have thought that this change would have a corresponding test change.  This suggests that we don't have any tests that check that the fields we are expecting are actually present.  This should be rectified.

**Comment by drothlis** [comment]
:+1: (assuming Travis is happy).

**Pull request merged**